packages/uicore/src/components/TableV2/TableV2.tsx

```diff
@@ -195,7 +195,11 @@ export const TableV2 = <Data extends Record<string, any>>(props: TableProps<Data
           )
         })}
       </div>
-      {pagination ? <Pagination className={css.pagination} {...pagination} /> : null}
+      {pagination ? (
+        pagination.itemCount > pagination.pageSize ? (
```

Comment:
nit: when we say exceed the page, we mean itemCount is greater than page size or visually it should not be overflowing the page?

Reply:
itemCount is greater than pageSize.